delivery_location/tests/test_deliverylocation.py

```diff
@@ -38,48 +33,45 @@ def test_filter_by_id_that_does_not_exist(self):
             DeliveryLocation.get_delivery_location_by_id(-1)
 
     @pytest.mark.django_db()
-    def test_filter_by_location_from_supplier(self, supplier0, delivery_location0):
-        supplier0.save()
+    def test_filter_by_location_from_supplier(self, delivery_location0):
         delivery_location0.add_delivery_location()
         assert delivery_location0 in list(
-            DeliveryLocation.filter_by_supplier_and_location(supplier0, "Kiryat Shemona"))
+            DeliveryLocation.filter_by_supplier_and_location(delivery_location0.user_name, "Kiryat Shemona"))
 
     @pytest.mark.django_db()
-    def test_filter_by_location_that_does_not_exist_from_supplier(self, supplier0):
-        supplier0.save()
-        assert [] == list(DeliveryLocation.filter_by_supplier_and_location(supplier0,
+    def test_filter_by_location_that_does_not_exist_from_supplier(self, saved_supplier0):
+        assert [] == list(DeliveryLocation.filter_by_supplier_and_location(saved_supplier0,
                                                                            "LOCATION THAT IS NOT RELATED TO SUPPLIER0"))
 
     @pytest.mark.django_db()
-    def test_filter_by_location(self, supplier0, delivery_location0, delivery_location1):
-        supplier0.save()
+    def test_filter_by_location(self, delivery_location0, delivery_location1):
+        # In here we are making both with the same location
         delivery_location0.add_delivery_location()
         delivery_location1.add_delivery_location()
+        delivery_location0.update_location("Haifa")
         assert set([delivery_location0, delivery_location1]) == set(list(
-            DeliveryLocation.filter_by_location(supplier0)))
+            DeliveryLocation.filter_by_location(delivery_location0.location)))
```

Comment:
i would update delivery location1 to show in the test that their location are truly both set to haifa